src/Simplex/Chat.hs

```diff
@@ -2057,6 +2035,22 @@ processChatCommand' vr = \case
   StopRemoteCtrl -> withUser_ $ stopRemoteCtrl >> ok_
   ListRemoteCtrls -> withUser_ $ CRRemoteCtrlList <$> listRemoteCtrls
   DeleteRemoteCtrl rc -> withUser_ $ deleteRemoteCtrl rc >> ok_
+  APIXFTPDirectUpload userId file -> withUserId userId $ \user -> do
+    fileSize <- liftIO $ CF.getFileContentsSize file
+    (_, _, fileTransferMeta) <- xftpSndFileTransfer_ user file fileSize 1 Nothing
+    pure CRSndFileStartXFTPDirect {user, fileTransferMeta}
+  APIXFTPDescriptionUpload userId fileTransferId -> withUserId userId $ \user -> do
```

Comment:
will UI call both APIs in turn? why not make it a single API that requests redirect from agent after original file is uploaded?